examples/native-fungible/src/contract.rs

```diff
@@ -0,0 +1,287 @@
+            Operation::Transfer {
+                owner,
+                amount,
+                target_account,
+            } => {
+                Self::check_account_authentication(context.authenticated_signer, owner)?;
+                let account_owner = owner;
+                let owner = self.normalize_owner(owner);
+
+                let fungible_target_account = target_account;
+                let target_account = self.normalize_account(target_account);
+
+                system_api::transfer(Some(owner), target_account, amount);
+
+                Ok(self.get_transfer_outcome(account_owner, fungible_target_account, amount))
+            }
+
+            Operation::Claim {
+                source_account,
+                amount,
+                target_account,
+            } => {
+                Self::check_account_authentication(
+                    context.authenticated_signer,
+                    source_account.owner,
+                )?;
+
+                let fungible_source_account = source_account;
+                let fungible_target_account = target_account;
+
+                let source_account = self.normalize_account(source_account);
+                let target_account = self.normalize_account(target_account);
+
+                system_api::claim(source_account, target_account, amount);
+
+                if fungible_source_account.chain_id == system_api::current_chain_id() {
+                    Ok(self.get_transfer_outcome(
+                        fungible_source_account.owner,
+                        fungible_target_account,
+                        amount,
+                    ))
+                } else {
+                    // If different chain, send message that will be ignored so the app gets auto-deployed
+                    let message = Message::Withdraw {
+                        owner: fungible_source_account.owner,
+                        amount,
+                        target_account: fungible_target_account,
+                    };
+                    Ok(ExecutionOutcome::default()
+                        .with_authenticated_message(fungible_source_account.chain_id, message))
+                }
+            }
+        }
+    }
+
+    async fn execute_message(
+        &mut self,
+        context: &MessageContext,
+        message: Self::Message,
+    ) -> Result<ExecutionOutcome<Self::Message>, Self::Error> {
+        // Messages for now don't do anything, just pass messages around
+        match message {
+            Message::Credit {
+                amount: _,
+                target: _,
+                source: _,
+            } => Ok(ExecutionOutcome::default()),
+            Message::Withdraw {
+                owner,
+                amount,
+                target_account,
+            } => {
+                Self::check_account_authentication(context.authenticated_signer, owner)?;
+                Ok(self.get_transfer_outcome(owner, target_account, amount))
```

Comment:
If the message is bounced, we should probably just ignore it.

Reply:
Please let's get to the bottom of this. Did I forget this case in `fungible` as well?

Reply:
As discussed in the meeting, since we're doing nothing here it doesn't matter if we're not taking bouncing credit messages into account. I'll add a comment though